File: node/python-net/udp_server.py
import socket

# 创建upd 套接字（socket.AF_INET代表ipv4，socket.SOCK_DGRAM代表使用upd协议）
s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)

# 绑定服务的网络信息
addr = ('', 6789)  # 注意，这里是元组类型，ip是字符串，端口号是数字。不写ip地址，程序在运行时会自动把ip设成本机ip地址。不绑定端口号，程序在运行时系统会为程序分配一个临时随机端口号
s.bind(addr)

# 循环接收对方发送来的数据（如果不循环，等到 在接收到对方一个数据后，程序就会结束运行）
while True:
    recv_data = s.recvfrom(1024) # 这里1024表示每次接收数据的最大字节数
    # 打印接收到的消息
    print('msg:',recv_data[0].decode(),'ip:',recv_data[1][0],'port:',recv_data[1][1])

# 关闭socket
s.close()

File: node/python-net/udp_server.js
var dgram = require('dgram');

//创建 udp server
var udp_server = dgram.createSocket('udp4');
udp_server.bind(5678); //为这个udp服务绑定端口号

//监听端口
udp_server.on('listening', function () {
    console.log('udp server linstening 5678');
})

//接收消息
udp_server.on('message', function (msg, rinfo) {
    strmsg = msg.toString();
    udp_server.send(strmsg, 0, strmsg.length, rinfo.port, rinfo.address); //将接收到的消息返回给客户端
    console.log(`udp server received data: ${strmsg} from ${rinfo.address}:${rinfo.port}`)
})

//错误处理
udp_server.on('error', function (err) {
    console.log('some error on udp server')
    udp_server.close();
})



//https://www.cnblogs.com/ay-a/p/9822268.html